### PR TITLE
Remove reference to blocktree from consensus TCB

### PR DESCRIPTION
**Pull request by weilei0107**
This PR removes the dependency on block tree from the consensus safety module.

### CODE REVIEW

File: consensus/src/chained_bft/safety/safety_rules.rs [hunk 1]
```diff
@@ -261,7 +239,7 @@ impl<T: Payload> SafetyRules<T> {
     /// sent).
     /// Requires that all the ancestors of the block are available for at least up to the last
     /// committed block, might panic otherwise.
-    pub fn voting_rule(
+    pub fn voting_rule<T: Payload>(
```
Comment (dmitri-perelman) [inline consensus/src/chained_bft/safety/safety_rules.rs]: It's interesting that in today's world we don't even need to receive the Block<T> here, but in future we might want to verify the `id` of a block corresponding to the other fields, hence let's keep it here for now (this is going to be the only function with a `<T>` parameter).

File: consensus/src/chained_bft/safety/safety_rules.rs [hunk 2]
```diff
@@ -209,36 +193,30 @@ impl<T: Payload> SafetyRules<T> {
...
+            self.state
+                .set_preferred_block_round(qc.certified_parent_block_round());
         }
     }
 
     /// Check if a one-chain at round r+2 causes a commit at round r and return the committed
-    /// block at round r if possible
-    pub fn commit_rule_for_certified_block(
+    /// block id at round r if possible
+    pub fn commit_rule_for_certified_block<T: Payload>(
```
Comment (dmitri-perelman) [inline consensus/src/chained_bft/safety/safety_rules.rs]: We can probably change the signature to receive just the QC.

File: consensus/src/chained_bft/safety/safety_rules.rs [hunk 3]
```diff
@@ -209,36 +193,30 @@ impl<T: Payload> SafetyRules<T> {
...
-                    return Some(three_chain_head);
-                }
-            }
+    ) -> Option<HashValue> {
+        // We're using a so-called 3-chain commit rule: B0 (as well as its prefix)
+        // can be committed if there exist certified blocks B1 and B2 that satisfy:
+        // 1) B0 <- B1 <- B2 <--
+        // 2) round(B0) + 1 = round(B1), and
+        // 3) round(B1) + 1 = round(B2).
+        let qc = one_chain_head.quorum_cert();
```
Comment (zekun000) [inline consensus/src/chained_bft/safety/safety_rules.rs]: can we rename this to be two_chain_qc or sth similar?

File: consensus/src/chained_bft/safety/safety_rules.rs (hunk 2 above)
Comment (zekun000) [reply consensus/src/chained_bft/safety/safety_rules.rs]: This can be a private function?

Comment (weilei0107) [reply consensus/src/chained_bft/safety/safety_rules.rs]: Made all the changes recommended. Now need another approve. :-)

**Pull request merged**